lib/ansible/plugins/lookup/onepassword_doc.py

```diff
@@ -0,0 +1,135 @@
+        version_added: '2.7'
+      secret_key:
+        description: The secret key used when performing an initial sign in.
+        version_added: '2.7'
+      vault:
+        description: Vault containing the item to retrieve (case-insensitive). If absent will search all vaults
+        default: None
+    notes:
+      - This lookup will use an existing 1Password session if one exists. If not, and you have already
+        performed an initial sign in (meaning C(~/.op/config exists)), then only the C(master_password) is required.
+        You may optionally specify C(subdomain) in this scenario, otherwise the last used subdomain will be used by C(op).
+      - This lookup can perform an initial login by providing C(subdomain), C(username), C(secret_key), and C(master_password).
+      - Due to the B(very) sensitive nature of these credentials, it is B(highly) recommeneded that you only pass in the minial credentials
+        needed at any given time. Also, store these credentials in an Ansible Vault using a key that is equal to or greater in strength
+        to the 1Password master password.
+      - This lookup stores potentially sensitive data from 1Password as Ansible facts.
+        Facts are subject to caching if enabled, which means this data could be stored in clear text
+        on disk or in a database.
+      - Tested with C(op) version 0.5.3
+"""
+
+EXAMPLES = """
+# These examples only work when already signed in to 1Password
+- name: Retrieve document contents for MyAwesomeDocument when already signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc', 'MyAwesomeDocument')
+
+- name: Retrieve document contents for MySecureTLSCert from a specific vault
+  debug:
+    var: lookup('onepassword_doc', 'MySecureTLSCert', vault='MyVault')
+
+- name: Retrieve MySecureTLSCert document when not signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password)
+
+- name: Retrieve MySecureTLSCert document when never signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password
+                username='tweety@acme.com'
+                secret_key=vault_secret_key)
+"""
+
+RETURN = """
+  _raw:
+    description: document contents requested
+"""
+
+from ansible.plugins.lookup.onepassword import OnePass
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_bytes
+
+
+class OnePassDoc(OnePass):
+
+    def get_document_raw(self, item_id, vault=None):
+        args = ["get", "document", item_id]
+        if vault is not None:
+            args += ['--vault={0}'.format(vault)]
+        if not self.logged_in:
+            args += [to_bytes('--session=') + self.token]
+        rc, output, dummy = self._run(args)
```

Comment:
Might want to check `rc` status before return output.

Reply:
Since I am subclassing `lib/ansible/plugins/lookup/onepassword.py`, `self._run` is defined there, and handles any return code besides `0`. See: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/onepassword.py#L191-L198, and raises an `AnsibleLookupError` if a non 0 return code.

Reply:
Sorry, my bad. I didn't see the implementation of `_run`.